src/Inciter/AMR/Refinement_State.h

```diff
@@ -0,0 +1,167 @@
+//
+// Created by Robert Francis Bird on 10/27/16.
+//
+
+#ifndef TETAMR_AMR_DATA_H
+#define TETAMR_AMR_DATA_H
+
+#include <vector>
+#include <cstddef>
+
+#include "types.h"
+
+namespace AMR {
+    // Types
+    using real_t = tk::real;
+
+    const size_t DEFUALT_CHILD_NUMBER = 0; // Used to mark "doesn't have children"
+    const size_t DEFAULT_NUM_CHILDREN = 0; // Default to no children
+
+    // TODO: Populate this enum with the available refinement cases
+    enum Refinement_Case { initial_grid, one_to_two, one_to_four, one_to_eight,
+        two_to_eight, four_to_eight, none };
+
+    enum Edge_Lock_Case {unlocked, locked, intermediate, temporary};
+
+    struct Edge_Refinement {
+        size_t A;
+        size_t B;
+        real_t refinement_criteria;
+        bool needs_refining; // TODO: This could possibly be deduced implicitly
+        bool needs_derefining; // TODO: Marge this with needs_refining
+        bool is_dead;
+        Edge_Lock_Case lockCase; // TODO: Refactor this to match _ style?
+
+        // Explicit Empty Constructor
+        Edge_Refinement() {
```

Comment:
That's what thought you meant though. So what is the reason for _not_ making this 'explicit'.